channel.go

```diff
@@ -418,6 +419,7 @@ It is safe to call this method multiple times.
 
 */
 func (ch *Channel) Close() error {
+	defer ch.connection.releaseChannel(ch.id)
```

Comment:
Should we release channel ID only after sending a `channel.close` and receiving a `channel.close-ok`? [This channel.close handler comment](https://github.com/rabbitmq/rabbitmq-server/blob/stable/src/rabbit_channel.erl#L890) to me suggests that before a client receives a `channel.close-ok`, it is not supposed to reuse the channel ID. Even if that's not the case, by releasing at the latest possible moment we reduce the probability of races for workloads with channel churn.